src/Api/Tools/Controllers/ImportCiphersController.cs

```diff
@@ -64,14 +74,46 @@ public async Task PostImport([FromBody] ImportCiphersRequestModel model)
         }
 
         var orgId = new Guid(organizationId);
-        if (!await _currentContext.AccessImportExport(orgId))
+        var collections = model.Collections.Select(c => c.ToCollection(orgId)).ToList();
+        var orgFlexibleCollections = (await _organizationService.GetByIdAsync(orgId)).FlexibleCollections;
+
+
+        //An User is allowed to import if CanCreate Collections or has AccessToImportExport
+        var authorized = orgFlexibleCollections
+            ? await CheckOrgImportPermission(collections, orgId) || await _currentContext.AccessImportExport(orgId)
+            : await _currentContext.AccessImportExport(orgId);
```

Comment:
This can be simplified by pushing the `_currentContext.AccessImportExport(orgId)`-check into `CheckOrgImportPermission(collections, orgId)` and using some early returns.